lib/Mojolicious/Plugin/OpenAPI/Guides/Security.pod

```diff
@@ -0,0 +1,110 @@
+=head1 NAME
+
+Mojolicious::Plugin::OpenAPI::Guides::Security - How to secure your API
+
+=head1 OVERVIEW
+
+This guide will give you an introduction on how to use the security features
+in OpenAPI, together with L<Mojolicious::Plugin::OpenAPI>.
+
+Note that this is currently EXPERIMENTAL! Please let me know if you have any
+feedback. See
+L<https://github.com/jhthorsen/mojolicious-plugin-openapi/pull/40> for a
+complete discussion.
+
+=head1 TUTORIAL
+
+=head2 Specification
+
+Here is an example specification that use
+L<securityDefinitions|http://swagger.io/specification/#securityDefinitionsObject>
+and L<security|http://swagger.io/specification/#securityRequirementObject> from
+the OpenAPI spec:
+
+  {
+    "swagger": "2.0",
+    "info": { "version": "0.8", "title": "Super secure" },
+    "schemes": [ "https" ],
+    "basePath": "/api",
+    "securityDefinitions": {
+      "dummy": {
+        "type": "apiKey",
+        "name": "Authorization",
+        "in": "header",
+        "description": "dummy"
+      }
+    },
+    "paths": {
+      "/protected": {
+        "post": {
+          "x-mojo-to": "super#secret_resource",
+          "security": [{"dummy": []}],
+          "parameters": [
+            { "in": "body", "name": "body", "schema": { "type": "object" } }
+          ],
+          "responses": {
+            "200": {"description": "Echo response", "schema": { "type": "object" }},
+            "401": {"description": "Sorry mate", "schema": { "type": "array" }}
+          }
+        }
+      }
+    }
+  }
+
+=head2 Application
+
+The specification above can be dispatched to handlers inside your
+L<Mojolicious> application. The do so, add the "security" key when loading the
+plugin, and reference the "securityDefinitions" name inside that to a callback.
+In this example, we have the "dummy" security handler:
+
+  package Myapp;
+  use Mojo::Base "Mojolicious";
+
+  sub startup {
+    my $app = shift;
+
+    $app->plugin(OpenAPI => {
+      url      => "data://main/sec.json",
+      security => {
+        dummy => sub {
+          my ($c, $definition, $scopes, $cb) = @_;
+          return $cb->(1) if $c->req->headers->authorization;
+          return $cb->(0);
```

Comment:
That is kinda what I was saying in my second paragraph [above](https://github.com/jhthorsen/mojolicious-plugin-openapi/pull/40#discussion_r125997847). In this case I guess we have them call `$c->$cb()` or `$c->$cb(undef)` to indicate no error (ie. pass). We would need to consider what they should return. I'd think probably an error string or else an array of error strings. Then if no success is given the result could be the standard error document where each failed condition is an error in the errors object (perhaps even with the path to the security check that failed).